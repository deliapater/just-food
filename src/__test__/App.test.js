import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import React from 'react';
import App from '../App';

afterEach(() => {
  cleanup();
})

describe('App tests', () => {
  it('should render App', () => {
    render(<App />);
    expect(App).toBeTruthy();
  });

  it('should contain toggle btn', () => {
    render(<App />);
    const toggleBtn = screen.getByText('Order Food');
    expect(toggleBtn).toBeTruthy()
  });

  it('should display title and subtitle', () => {
    render(<App />);
    const title =  screen.getByText('Just Food Online Shop');
    const subTitle =  screen.getByText('Menu Availability');
    expect(title).toBeTruthy();
    expect(subTitle).toBeTruthy();
    expect(title).toBeInTheDocument();
  })

  it('should contain food elements', () => {
    render(<App/>);
    const foodEl = screen.getByTestId('food-el');
    expect(foodEl).toHaveTextContent('Arepa');
  })
})

describe('should match snapshot', () => {
  const tree = renderer.create(<App/>).toJSON();
  expect(tree).toMatchSnapshot();
})
