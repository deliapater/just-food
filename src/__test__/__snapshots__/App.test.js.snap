// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="App"
>
  <button
    className="toggleButton"
    onClick={[Function]}
  >
    Order Food
  </button>
  <h3
    className="title"
  >
    Just Food Online Shop
  </h3>
  <h4
    className="subTitle"
  >
    Menu Availability
  </h4>
  <ul
    className="ulApp"
    data-testid="food-el"
  >
    <li
      className="liApp"
    >
      Chicken Burger
       - 
      40
    </li>
    <li
      className="liApp"
    >
      Veg Burger
       - 
      30
    </li>
    <li
      className="liApp"
    >
      Chips
       - 
      50
    </li>
    <li
      className="liApp"
    >
      Ice Cream
       - 
      30
    </li>
    <li
      className="liApp"
    >
      Arepa
       - 
      33
    </li>
  </ul>
</div>
`;
